### PR TITLE
Add blog: Three years at Sourcegraph as a software engineer

### PR DESCRIPTION
**Pull request by jdorfman**
### Changelog

- Add new blog post by Joe Chen

### Test

1. Ensure prettier has standardized the proposed changes.

### CODE REVIEW

**Comment by unknwon** [comment]
Hmm, looks like this isn't converted from the [copy-edited version](https://docs.google.com/document/d/1C6lJK9bX0WCUNvNDRmOEiCc0DNLVsOsBho_Gs493IjU/edit#heading=h.wvysjipmifbc) but the [original version](https://docs.google.com/document/d/1AIr_oxhTTYI3Axh5VsF6uSY_-h2q5sRcJJeASpelFS8/edit#heading=h.wvysjipmifbc) (which I posted on my personal blog), is this intended?

**Comment by jdorfman** [comment]
@unknwon that is my fault, I will fix it now.

File: content/blogposts/2022/three-years-at-sourcegraph.md [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+
+As Quinn used to say, “It’s a marathon not a sprint”, and you need to be sustainable.
+
+### “Better to ask for forgiveness than permission”
+
+I first heard this saying from my (previous) manager [Rafal Leszczynski](https://www.linkedin.com/in/rleszcz/). I think it’s a key idea to remember to act at a reasonable speed with high agency. You can’t just wait for things to happen, especially while being accountable.
+
+Come up with a communication plan, pull in stakeholders, provide context, be clear about the things to ask for, set expectations, and propose a deadline. Reaching consensus isn’t always possible, but it’s important to make people feel heard with their feedback.
+
+Individuals and teams make decisions all the time. While moving fast, be mindful of [reversible and irreversible decisions](https://fs.blog/reversible-irreversible-decisions/#:~:text=Reversible%20decisions%20are%20doors%20that,the%20invested%20time%20and%20resources).). If something isn’t reversible, or the cost is too high to reverse it, it doesn’t hurt to give it a second review before getting buy-in from leadership.
```
Comment (unknwon) [inline content/blogposts/2022/three-years-at-sourcegraph.md:147]:
```suggestion
Individuals and teams make decisions all the time. While moving fast, be mindful of [reversible and irreversible decisions](https://fs.blog/reversible-irreversible-decisions/#:~:text=Reversible%20decisions%20are%20doors%20that,the%20invested%20time%20and%20resources). If something isn’t reversible, or the cost is too high to reverse it, it doesn’t hurt to give it a second review before getting buy-in from leadership.
```

Comment (unknwon) [reply content/blogposts/2022/three-years-at-sourcegraph.md:147]: CleanShot 2022-07-28 at 00 44 02@2x

File: content/blogposts/2022/three-years-at-sourcegraph.md [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+
+### Organization hierarchy
+
+We just reorged (yet again within a year), and our latest [org chart](https://handbook.sourcegraph.com/team/org_chart/) is documented in the handhook. In the Engineering department, we had multiple organizations with one director and one group PM, then we had teams consisting of one EM, one PM, one product designer and multiple ICs (Note: Not all teams are fully staffed.)
+
+The product designer on my team (IAM) had been loaned to form a tiger team before the reorg. We had just lost our EM and one IC to form a new team under the Cloud department, and our PM had stepped up to be the group PM of the Admin Experience organization. What a sad story! 😅Is this the “darkest” moment I’ve experienced here? Not quite, there was a brief period where I was the only one left on the team.
+
+### Project management
+
+#### The curse of GitHub
```
Comment (unknwon) [inline content/blogposts/2022/three-years-at-sourcegraph.md:199]: This is not related to the content, but a general bug in our blog I think, where heading 4 looks bigger than heading 3

CleanShot 2022-07-28 at 00 48 36@2x

File: content/blogposts/2022/three-years-at-sourcegraph.md [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+
+(Aside: Wrong scripts!)
+
+Sourcegraph is an inspiring place to work, with truly amazing, empathetic and thoughtful leadership and teammates. I’m looking forward to continuing making contributions and having a greater impact for many years to come.
+
+It’s a place where I see most people willing to spend years for their career. Of course, I have my own reasons for working, and for transparency, I give them away honestly:
+
+* I have faith in the mission: making it so “everyone can code” and realizing “being able to code will become a basic life skill, like reading was centuries ago”.
+
+* There’s probably no better place to be surrounded by talented people and working with them pleasantly
```
Comment (unknwon) [inline content/blogposts/2022/three-years-at-sourcegraph.md:388]:
```suggestion
* There’s probably no better place to be surrounded by talented people and working with them pleasantly.
```

**Review by unknwon — APPROVED** [review]

**Review by unknwon — APPROVED** [review]

**Pull request merged**